src/Microsoft.Kusto.ServiceLayer/DataSource/KustoDataSource.cs

```diff
@@ -347,38 +347,43 @@ private IEnumerable<DataSourceObjectMetadata> GetDatabaseMetadata(bool includeSi
         {
             if (_databaseMetadata == null)
             {
-                CancellationTokenSource source = new CancellationTokenSource();
```

Comment:
Moved all this logic into `SetDatabaseMetadata` function

Reply:
SetDatabaseMetadata is doing the same thing. Is it being extracted for testing purposes? If you need to extract it out, the name is a little misleading too. Maybe call it InternalGetMetadata. We are not setting any database info on Kusto.

Reply:
I extracted it because I'm using the 'Set' functionality in the Refresh method. I have the Refresh method simply resetting the data but not returning it.